moto/redshiftdata/models.py

```diff
@@ -0,0 +1,192 @@
+        self.ResultSize = -1
+        self.SecretArn = secret_arn
+        self.Status = 'STARTED'
+        self.SubStatements = []
+        self.UpdatedAt = now
+
+
+class StatementResult:
+    def __init__(
+        self,
+        column_metadata,
+        records,
+        total_number_rows,
+        next_token=None,
+    ):
+        self.ColumnMetadata = column_metadata
+        self.Records = records
+        self.NextToken = next_token
+        self.TotalNumberRows = total_number_rows
+
+
+class ColumnMetadata:
+    def __init__(
+        self,
+        column_default,
+        is_case_sensitive,
+        is_signed,
+        name,
+        nullable
+    ):
+        self.columnDefault = column_default
+        self.isCaseSensitive = is_case_sensitive
+        self.isSigned = is_signed
+        self.name = name
+        self.nullable = nullable
+
+
+class Record:
+    def __init__(
+        self,
+        **kwargs,
+    ):
+        if "long_value" in kwargs:
+            self.longValue = kwargs["long_value"]
+        elif "string_value" in kwargs:
+            self.stringValue = kwargs["string_value"]
+
+
+class RedshiftDataAPIServiceBackend(BaseBackend):
+
+    def __init__(self, region_name=None):
+        self.region_name = region_name
+        self.statements = {}
+
+    def reset(self):
+        region_name = self.region_name
+        self.__dict__ = {}
+        self.__init__(region_name)
+
+    def cancel_statement(self, statement_id):
+        _validate_uuid(statement_id)
+
+        try:
+            # Statement exists
+            statement = self.statements[statement_id]
+
+            if statement.Status != "STARTED":
+                raise ValidationException(
+                    "Could not cancel a query that is already in %s state with ID: %s"
+                    % (statement.Status, statement_id)
+                )
+
+            statement.Status = "ABORTED"
+            self.statements[statement_id] = statement
+        except KeyError:
+            # Statement does not exist.
+            raise ResourceNotFoundException()
+
+        return True
+
+    def describe_statement(self, statement_id):
+        _validate_uuid(statement_id)
+
+        try:
+            # Statement exists
+            return self.statements[statement_id]
+        except KeyError:
+            # Statement does not exist.
+            raise ResourceNotFoundException()
+
+    def execute_statement(
+        self,
+        cluster_identifier,
+        database,
+        db_user,
+        parameters,
+        secret_arn,
+        sql,
+    ):
+        statement = Statement(
+            cluster_identifier=cluster_identifier,
+            database=database,
+            db_user=db_user,
+            query_parameters=parameters,
+            query_string=sql,
+            secret_arn=secret_arn
+        )
+        self.statements[statement.Id] = statement
+        return statement
+
+    def get_statement_result(self, statement_id):
```

Comment:
Can we put an explanation on what this method here, as part of the docstring?
```
def get_statement_result(..)
"""
Return static statement result, because ..

"""
```
Setting the docstring ensures that the explanation automatically becomes part of the docs when calling `update_implementation_coverage`

Reply:
Good point. I added the docstring